save-plugins/warn-plugin/src/commonMain/kotlin/org/cqfn/save/plugin/warn/WarnPlugin.kt

```diff
@@ -44,11 +42,19 @@ class WarnPlugin(
         testConfig.validateAndSetDefaults()
 
         val warnPluginConfig = testConfig.pluginConfigs.filterIsInstance<WarnPluginConfig>().single()
-        val generalConfig = testConfig.pluginConfigs.filterIsInstance<GeneralConfig>().singleOrNull()
+        val generalConfig = testConfig.pluginConfigs.filterIsInstance<GeneralConfig>().single()
 
-        return files.chunked(warnPluginConfig.batchSize ?: 1).map { chunk ->
-            handleTestFile(chunk.map { it.single() }, warnPluginConfig, generalConfig)
-        }.flatten()
+        // Special trick to handle cases when tested tool is able to process directories.
+        // In this case instead of executing the tool with file names, we execute the tool with directories.
+        //
+        // In case user wants to use directory mode, he needs simply not to pass [wildCardInDirectoryMode] and it will be null
```

Comment:
?
```suggestion
        // In case, when user doesn't wants to use directory mode, he needs simply not to pass [wildCardInDirectoryMode] and it will be null
```